### PR TITLE
use lock_guard to protect the access to m_exec_map in cpu_backend

### PR DESCRIPTION
**Pull request by pruthviIntel**

### CODE REVIEW

**Comment by pruthviIntel** [comment]
@pawelpiotrowicz just wondering if you were able to cross check against the model which used to segfault for multi threaded instance?

File: src/ngraph/runtime/cpu/cpu_backend.cpp [hunk 1]
```diff
@@ -90,6 +90,7 @@ shared_ptr<runtime::Executable>
                                        ngraph::pass::PassConfig& pass_config,
                                        bool performance_counters_enabled)
 {
+    std::lock_guard<std::mutex> guard(m_exec_map_mutex);
```
Comment (jbobba) [inline src/ngraph/runtime/cpu/cpu_backend.cpp]: we only want to guard the access and modifications to m_exec_map. make_shared<CPU_Executable>(func, pass_config, performance_counters_enabled) should be outside the critical section.

**Comment by pawelpiotrowicz** [comment]
> @pawelpiotrowicz just wondering if you were able to cross check against the Paddle-Paddle model which used to segfault for multi threaded instance?

@pruthviIntel  I'm not sure, could you clarify the question? an example ?

File: src/ngraph/runtime/cpu/cpu_backend.cpp (hunk 1 above)
Comment (pruthviIntel) [reply src/ngraph/runtime/cpu/cpu_backend.cpp]: @jbobba i addressed the changes requested , can u check?

File: src/ngraph/runtime/cpu/cpu_backend.cpp [hunk 2]
```diff
@@ -91,17 +91,23 @@ shared_ptr<runtime::Executable>
...
+    // m_exec_map_mutex will be released once the object `guard` goes out of scope
     {
-        rc = it->second;
+        std::lock_guard<std::mutex> guard(m_exec_map_mutex);
+        auto it = m_exec_map.find(func);
+        if (it != m_exec_map.end())
+        {
+            rc = it->second;
+        }
+        return rc;
```
Comment (nmostafa) [inline src/ngraph/runtime/cpu/cpu_backend.cpp]: That return should be inside the if, no ?

Comment (pruthviIntel) [reply src/ngraph/runtime/cpu/cpu_backend.cpp]: ah right!. thanks for catch it.

**Pull request merged**